npartovi/MTGCG
# Create loading spinner

## Description
This PR adds the appropriate actions, reducers, and components needed to create a loading spinner. For now, this loading spinner is large, but if needed we can add a "size" prop to the loading spinner that allows someone to specify various sizes for the loading spinner.

Other changes made: 
* Moved mtgFilter and corresponding types into its own folder
* Prettified a few things

## Test Plan
* Load localhost:3000.
* Select a few options to the right and click 'Generate Cards'. There should be a large loading spinner that appears on the screen for a moment while the cards are being fetched.

## Regression Test Plan
* Everything should still work and function as before (selecting colors from the left, generating cards, etc.).

## Note on merging
The person reviewing this PR should 'Squash and merge' this PR, so as to keep unnecessary commits to a minimum 😇
